<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Critical 1Cloud Server Vulnerability</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="We want to share our experience with 1cloud servers. We do not want to blame anyone, just want to put the problem on public display, so that 1cloud em...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Critical 1Cloud Server Vulnerability</h1><div class="post__text post__text-html js-mediator-article">  We want to share our experience with 1cloud servers.  We do not want to blame anyone, just want to put the problem on public display, so that 1cloud employees will have the motivation to conduct a detailed analysis of this problem. <br><br>  It all started with the fact that the last months on the Windows 1cloud servers, we began to notice an abnormal load, but did not attach any special importance to this, since the Windows server was mainly used to perform various small tasks such as using the browser and any other programs, and especially there was a desire to find out what was there and how. <br><br>  But the other day there was a very unpleasant and insulting situation on the part of 1cloud employees, which pushed us to publish this article.  We have been using 1cloud servers for about 1.5-2x years, the rules for using servers have not been violated.  I was unpleasantly surprised when I logged into my account and saw that all 40 of our servers were blocked with the note ‚ÄúServer is blocked by the administrator.  Reason for blocking: brute force. ‚Äù <a name="habracut"></a>  I was going to write to the support and saw a new message from the technical support service, which said: <br><blockquote>  <i>Hello.</i> <i>
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    </i>  <i>Anomalous network activity was detected from your IP 111.111.11.111 server: attempts to connect to a large number of arbitrary servers on port 22 (SSH)</i> <i><br></i>  <i>The server may have been hacked.</i>  <i>You need to quickly fix the problem.</i> <i><br><br></i>  <i>If there is no response on this request, the server will be disconnected from the network.</i> </blockquote><br>  Due to an employee error or due to any problems with the server management system, all 40 servers were turned off instead of one problem one, and they were unavailable for 24 hours.  From the support message was received as follows: <br><blockquote>  <i>Servers unlocked, they will turn on automatically.</i>  <i>According to internal instructions, one problem server should have been disconnected, this incident will be investigated.</i> <i><br></i>  <i>Please, in order to avoid such situations, respond to the appeal more quickly.</i> <i><br></i>  <i>Also please report upon the elimination of problems with the server.</i> </blockquote><br>  This is certainly an unpleasant situation, but it happens, I thought, and we began to understand the problem that caused the servers to be blocked.  The servers did not automatically turn on, as we were promised, and I had to manually start all the servers, but this is minor compared to the main problem.  Logged in to the server, and immediately noticed the oddity that ssh we connected the last time on August 20th, and various malicious files were uploaded to the server on the 21st and 22nd.  As it turned out, during the creation of each server on 1cloud, some doubtful user ‚Äúuser‚Äù is added, in the folder of which a bunch of malicious files later appeared in the miner, including the xrig miner. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/00d/dd6/900/00ddd6900940baa8d9596159be6b1d5e.png" alt="image"><br><br>  They started calling the support service, explained the situation, asked what user ‚Äúuser‚Äù was added to when creating the server and what malicious files were doing in its folders.  The tech support employee tried to deny that the user ‚Äúuser‚Äù was added automatically when creating the server, and claimed that they had nothing to do with it.  We created a new server for the test, on which the user ‚Äúuser‚Äù was also immediately present, but so far without malicious files, since the malicious software is injected into the server not immediately, but after a day or more.  The tech support employee could not give a clear answer, from where the user ‚Äúuser‚Äù appeared on the newly created server by default. <br><br>  It turned out to look at the logs, it turned out that someone, knowing the password, connected to the user ‚Äúuser‚Äù, and uploaded malicious files for mining and brute-force to the server.  At once I would like to clarify that 1cloud clients are not given a password from the user ‚Äúuser‚Äù, moreover, the technical support officer persistently argued to me that even such a user was not on their servers and never had, although we saw such users on all servers throughout the entire period use of servers. <br><br>  We believe, and everything indicates that one of the employees of 1cloud (I really hope that it is not a guide), began to do black deeds, installing miners and malicious software on the servers of their clients, for which, by the way, we pay money.  In the first days after the creation of the server, someone connects to it through the user ‚Äúuser‚Äù, knowing his password, and gains control over your server.  Perhaps this is a brute, you say, but no, unfortunately, the logs show that it is not a brute, namely, ordinary authorization without selecting passwords.  Before logging into the server as user ‚Äúuser‚Äù, the logs show an attempt to connect user, admin, ubuntu, ubnt, test and osmc to one-time users, and then successfully authorize through user ‚Äúuser‚Äù. <br><br> <code>Aug 20 03:46:59 debian8x64 sshd[1328]: Invalid user test from 219.135.136.144 <br> Aug 20 03:46:59 debian8x64 sshd[1328]: input_userauth_request: invalid user test [preauth] <br> Aug 20 03:46:59 debian8x64 sshd[1328]: pam_unix(sshd:auth): check pass; user unknown <br> Aug 20 03:46:59 debian8x64 sshd[1328]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=219.135.136.144 <br> Aug 20 03:47:02 debian8x64 sshd[1328]: Failed password for invalid user test from 219.135.136.144 port 1072 ssh2 <br> Aug 20 03:47:02 debian8x64 sshd[1328]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:03 debian8x64 sshd[1330]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:03 debian8x64 sshd[1330]: Invalid user debian from 219.135.136.144 <br> Aug 20 03:47:03 debian8x64 sshd[1330]: input_userauth_request: invalid user debian [preauth] <br> Aug 20 03:47:03 debian8x64 sshd[1330]: pam_unix(sshd:auth): check pass; user unknown <br> Aug 20 03:47:03 debian8x64 sshd[1330]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=219.135.136.144 <br> Aug 20 03:47:05 debian8x64 sshd[1330]: Failed password for invalid user debian from 219.135.136.144 port 8178 ssh2 <br> Aug 20 03:47:05 debian8x64 sshd[1330]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:06 debian8x64 sshd[1332]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:07 debian8x64 sshd[1332]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=219.135.136.144 user=root <br> Aug 20 03:47:09 debian8x64 sshd[1332]: Failed password for root from 219.135.136.144 port 14224 ssh2 <br> Aug 20 03:47:09 debian8x64 sshd[1332]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:10 debian8x64 sshd[1334]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:10 debian8x64 sshd[1334]: Invalid user debian from 219.135.136.144 <br> Aug 20 03:47:10 debian8x64 sshd[1334]: input_userauth_request: invalid user debian [preauth] <br> Aug 20 03:47:11 debian8x64 sshd[1334]: pam_unix(sshd:auth): check pass; user unknown <br> Aug 20 03:47:11 debian8x64 sshd[1334]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=219.135.136.144 <br> Aug 20 03:47:13 debian8x64 sshd[1334]: Failed password for invalid user debian from 219.135.136.144 port 21466 ssh2 <br> Aug 20 03:47:13 debian8x64 sshd[1334]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:15 debian8x64 sshd[1336]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:15 debian8x64 sshd[1336]: Invalid user osmc from 219.135.136.144 <br> Aug 20 03:47:15 debian8x64 sshd[1336]: input_userauth_request: invalid user osmc [preauth] <br> Aug 20 03:47:15 debian8x64 sshd[1336]: pam_unix(sshd:auth): check pass; user unknown <br> Aug 20 03:47:15 debian8x64 sshd[1336]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=219.135.136.144 <br> Aug 20 03:47:16 debian8x64 sshd[1336]: Failed password for invalid user osmc from 219.135.136.144 port 28516 ssh2 <br> Aug 20 03:47:17 debian8x64 sshd[1336]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:18 debian8x64 sshd[1338]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:18 debian8x64 sshd[1338]: Invalid user ubnt from 219.135.136.144 <br> Aug 20 03:47:18 debian8x64 sshd[1338]: input_userauth_request: invalid user ubnt [preauth] <br> Aug 20 03:47:18 debian8x64 sshd[1338]: pam_unix(sshd:auth): check pass; user unknown <br> Aug 20 03:47:18 debian8x64 sshd[1338]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=219.135.136.144 <br> Aug 20 03:47:20 debian8x64 sshd[1338]: Failed password for invalid user ubnt from 219.135.136.144 port 34656 ssh2 <br> Aug 20 03:47:20 debian8x64 sshd[1338]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:22 debian8x64 sshd[1340]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:22 debian8x64 sshd[1340]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=219.135.136.144 user=root <br> Aug 20 03:47:24 debian8x64 sshd[1340]: Failed password for root from 219.135.136.144 port 40882 ssh2 <br> Aug 20 03:47:24 debian8x64 sshd[1340]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:25 debian8x64 sshd[1342]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:25 debian8x64 sshd[1342]: Invalid user admin from 219.135.136.144 <br> Aug 20 03:47:25 debian8x64 sshd[1342]: input_userauth_request: invalid user admin [preauth] <br> Aug 20 03:47:26 debian8x64 sshd[1342]: Failed none for invalid user admin from 219.135.136.144 port 47736 ssh2 <br> Aug 20 03:47:26 debian8x64 sshd[1342]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:27 debian8x64 sshd[1344]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:27 debian8x64 sshd[1344]: Invalid user test from 219.135.136.144 <br> Aug 20 03:47:27 debian8x64 sshd[1344]: input_userauth_request: invalid user test [preauth] <br> Aug 20 03:47:27 debian8x64 sshd[1344]: pam_unix(sshd:auth): check pass; user unknown <br> Aug 20 03:47:27 debian8x64 sshd[1344]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=219.135.136.144 <br> Aug 20 03:47:29 debian8x64 sshd[1344]: Failed password for invalid user test from 219.135.136.144 port 50546 ssh2 <br> Aug 20 03:47:29 debian8x64 sshd[1344]: Connection closed by 219.135.136.144 [preauth] <br> Aug 20 03:47:30 debian8x64 sshd[1346]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> <b>Aug 20 03:47:31 debian8x64 sshd[1346]: Accepted password for user from 219.135.136.144 port 56492 ssh2</b> <br> Aug 20 03:47:31 debian8x64 sshd[1346]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:33 debian8x64 sshd[1350]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> <b>Aug 20 03:47:33 debian8x64 sshd[1350]: Accepted password for user from 219.135.136.144 port 60134 ssh2</b> <br> Aug 20 03:47:33 debian8x64 sshd[1350]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:35 debian8x64 sshd[1354]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> <b>Aug 20 03:47:35 debian8x64 sshd[1354]: Accepted password for user from 219.135.136.144 port 4485 ssh2</b> <br> Aug 20 03:47:35 debian8x64 sshd[1354]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:38 debian8x64 sshd[1358]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> <b>Aug 20 03:47:38 debian8x64 sshd[1358]: Accepted password for user from 219.135.136.144 port 9509 ssh2</b> <br> Aug 20 03:47:38 debian8x64 sshd[1358]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:42 debian8x64 sshd[1362]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:42 debian8x64 sshd[1362]: Accepted password for user from 219.135.136.144 port 15833 ssh2 <br> Aug 20 03:47:42 debian8x64 sshd[1362]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:43 debian8x64 sshd[1346]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:47:44 debian8x64 sshd[1366]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:44 debian8x64 sshd[1366]: Accepted password for user from 219.135.136.144 port 19619 ssh2 <br> Aug 20 03:47:44 debian8x64 sshd[1366]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:46 debian8x64 sshd[1370]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:46 debian8x64 sshd[1370]: Accepted password for user from 219.135.136.144 port 23935 ssh2 <br> Aug 20 03:47:46 debian8x64 sshd[1370]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:48 debian8x64 sshd[1374]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:49 debian8x64 sshd[1374]: Accepted password for user from 219.135.136.144 port 28277 ssh2 <br> Aug 20 03:47:49 debian8x64 sshd[1374]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:51 debian8x64 sshd[1378]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:51 debian8x64 sshd[1378]: Accepted password for user from 219.135.136.144 port 31735 ssh2 <br> Aug 20 03:47:51 debian8x64 sshd[1378]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:53 debian8x64 sshd[1383]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:53 debian8x64 sshd[1383]: Accepted password for user from 219.135.136.144 port 36097 ssh2 <br> Aug 20 03:47:53 debian8x64 sshd[1383]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:47:55 debian8x64 sshd[1388]: reverse mapping checking getaddrinfo for 144.136.135.219.broad.gz.gd.dynamic.163data.com.cn [219.135.136.144] failed - POSSIBLE BREAK-IN ATTEMPT! <br> Aug 20 03:47:56 debian8x64 sshd[1388]: Accepted password for user from 219.135.136.144 port 39885 ssh2 <br> Aug 20 03:47:56 debian8x64 sshd[1388]: pam_unix(sshd:session): session opened for user user by (uid=0) <br> Aug 20 03:48:16 debian8x64 sshd[1350]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:19 debian8x64 sshd[1354]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:22 debian8x64 sshd[1358]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:25 debian8x64 sshd[1362]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:27 debian8x64 sshd[1366]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:29 debian8x64 sshd[1370]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:31 debian8x64 sshd[1374]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:34 debian8x64 sshd[1378]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:36 debian8x64 sshd[1383]: pam_unix(sshd:session): session closed for user user <br> Aug 20 03:48:38 debian8x64 sshd[1388]: pam_unix(sshd:session): session closed for user user</code> <br> <br>  One conclusion from this is that 1cloud servers are currently unsafe to use for hosting any important projects, since your data may be lost due to the fact that some mole has wound up in 1cloud, which has left a hole in the OS templates.  At the moment, we managed to find out that this vulnerability is definitely present in the Debian 8 template, let the rest of the employees of 1cloud check themselves, we have already done enough to inform them about this vulnerability. <br><br>  After lengthy telephone conversations with 1cloud, we could not get a clear explanation of what was happening.  They reluctantly apologized to us and said that they would sometime be engaged in replacing the OS templates, and eliminate everything superfluous from them.  The question is being imposed, guys, what extra you need to remove from OS templates?  Do you completely spit on your customers?  How was it possible to sell servers with a similar gap to thousands of clients who store terabytes of important data on these servers?  Only one output ‚Äì1cloud appreciates only profit, customer data does not represent any value for them!  The guys from 1cloud, if you think that I'm wrong - prove me the opposite! <br><br>  We are open to communication, dear employees 1cloud, if you have any questions - write a ticket.  We are ready to discuss this situation, but only with someone competent, and not with an employee of the support team, who persistently tried to prove to us that you do not add any users with the name ‚Äúuser‚Äù when creating the server.  We are also ready to provide all the necessary data to help end the sale of vulnerable 1cloud servers once and for all! </div><p>Source: <a href="https://habr.com/ru/post/421131/">https://habr.com/ru/post/421131/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../421121/index.html">Streaming video through a browser with ultra-low latency (and WebRTC!)</a></li>
<li><a href="../421123/index.html">IT events digest for September</a></li>
<li><a href="../421125/index.html">We segment 600 million users in real time every day.</a></li>
<li><a href="../421127/index.html">Friday webinars from Skillbox: development and developers</a></li>
<li><a href="../421129/index.html">How to reduce code review from two weeks to several hours. Experience team Yandex. Market</a></li>
<li><a href="../421133/index.html">LINK. Paper keyboard. Very large buttons</a></li>
<li><a href="../421135/index.html">Au / Ni / MgO: heat transfer at the nanometer level</a></li>
<li><a href="../421137/index.html">How Headless Chrome Works</a></li>
<li><a href="../421139/index.html">Frontend Conf - with user care</a></li>
<li><a href="../421141/index.html">Mutational analysis, or how to test tests</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>